src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.MultiSegment.cs

```diff
@@ -1329,7 +1340,8 @@ private ConsumeNumberResult ConsumeNegativeSignMultiSegment(ref ReadOnlySpan<byt
                         }
                         return ConsumeNumberResult.NeedMoreData;
                     }
-                    _totalConsumed++;
+                    Debug.Assert(i == 1);
+                    _totalConsumed += i;
```

Comment:
Why += i rather than ++ if we guarantee i == 1?

Similarly, earlier on, there's a `i >= data.Length` check... should that just be `data.Length == 1`? data.Length can't be 0 there, and i can't be > 1 (if the asserts are to be believed).

Reply:
The original bug came about because in certain methods I was doing `_totalConsumed++` because I had reasoned about i being 1. However, I was mistaken and `_totalConsumed += i` was correct (for all cases).  So, for consistency, I decided to keep this the same and add a `Debug.Assert` since it is the only case where i is guaranteed to be 1 (hence the `Debug.Assert`). I can change it to `++` if you still think the consistency isn't worth it.